library/src/cache.rs

```diff
@@ -121,10 +128,21 @@ impl UpdaterState {
         let reader = BufReader::new(file);
         // TODO: Now that we depend on serde_yaml for shorebird.yaml
         // we could use yaml here instead of json.
-        let state = serde_json::from_reader(reader)?;
+        let mut state: UpdaterState = serde_json::from_reader(reader)?;
+        if state.client_id.is_none() {
+            // Generate a client id if we don't already have one.
+            state.client_id = Some(generate_client_id());
+            let _ = state.save();
+        }
         Ok(state)
     }
 
+    fn create_and_save_new(cache_dir: &Path, release_version: &str) -> Self {
```

Comment:
```suggestion
    fn new_and_save(cache_dir: &Path, release_version: &str) -> Self {
```
nit

Reply:
I tried to follow the naming pattern that exists in the `UpdaterState`. There's a lot of verb-noun naming for functions that do more than retrieve values (`validate_slot`, `activate_latest_bootable_patch``set_next_boot_patch_slot`, etc.), so I updated this to `create_new_and_save` to keep the `create` verb in there.